src/compiler/tools/LanguageService.ts

```diff
@@ -11,6 +11,18 @@ import { Program } from "./Program";
 import { DefinitionInfo, EmitOutput, FileTextChanges, ImplementationLocation, RenameLocation, TextChange, DiagnosticWithLocation, RefactorEditInfo, CodeFixAction,
     CombinedCodeActions } from "./results";
 
+export interface CompilerHostOverrides {
+    resolveModuleNames?(moduleNames: string[], containingFile: string, reusedNames?: string[], ref?: ts.ResolvedProjectReference): (ts.ResolvedModule | undefined)[];
+    getResolvedModuleWithFailedLookupLocationsFromCache?(modulename: string, containingFile: string): ts.ResolvedModuleWithFailedLookupLocations | undefined;
+    resolveTypeReferenceDirectives?(typeDirectiveNames: string[], containingFile: string, ref?: ts.ResolvedProjectReference): (ts.ResolvedTypeReferenceDirective | undefined)[];
+}
+
+export type CompilerHostFactory = (languageService: FileSystemHost) => CompilerHostOverrides;
+
+export interface LanguageServiceOptions {
```

Comment:
We could mark this with `/** @internal */` since the constructor is private in the declaration file.

Reply:
When you're done with all these changes, run the command `yarn overwrite-declaration-files`. It will update the declaration file stored in the repo (then check that in). Thanks!